backend/src/main/java/org/eclipse/tractusx/puris/backend/controller/PartnerController.java

```diff
@@ -0,0 +1,188 @@
+    @Autowired
+    private PartnerService partnerService;
+    private ModelMapper modelMapper = new ModelMapper();
+
+    @CrossOrigin
+    @PostMapping
+    @Operation(description = "Creates a new Partner entity with the data given in the request body. Please not that no " +
+        "UUID can be assigned to a Partner that wasn't created before. So the request body must not contain a UUID.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Partner created successfully."),
+        @ApiResponse(responseCode = "400", description = "Request body was malformed, didn't meet the minimum constraints or wrongfully contained a UUID."),
+        @ApiResponse(responseCode = "409", description = "The BPNL specified in the request body is already assigned. ")
+    })
+    public ResponseEntity<?> createPartner(@RequestBody PartnerDto partnerDto) {
+        modelMapper.getConfiguration().setPropertyCondition(Conditions.isNotNull());
+        // Any given UUID is wrong by default since we're creating a new Partner entity
+        if (partnerDto.getUuid() != null || partnerDto.getBpnl() == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+
+        // Check whether the given BPNL is already assigned
+        Partner checkExistingPartner = partnerService.findByBpnl(partnerDto.getBpnl());
+        if (checkExistingPartner != null) {
+            // Cannot create Partner because BPNL is already assigned
+            return new ResponseEntity<>(HttpStatusCode.valueOf(409));
+        }
+
+        Partner partnerEntity;
+        try {
+            partnerEntity = modelMapper.map(partnerDto, Partner.class);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+
+        partnerEntity = partnerService.create(partnerEntity);
+        if (partnerEntity == null) {
+            // Creation failed due to unfulfilled constraints
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        return new ResponseEntity<>(HttpStatusCode.valueOf(200));
+    }
+
+    @PutMapping("putAddress")
+    @CrossOrigin
+    @Operation(description = "Updates an existing Partner by adding a new Address. If that this Partner already has " +
+        "an Address with the BPNA given in the request body, that existing Address will be overwritten. ")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Update accepted"),
+        @ApiResponse(responseCode = "400", description = "Invalid Address data"),
+        @ApiResponse(responseCode = "404", description = "Partner not found"),
+        @ApiResponse(responseCode = "500", description = "")
+    })
+    public ResponseEntity<?> addAddress(
+        @Parameter(name = "partnerBpnl", description = "The unique BPNL that was assigned to that Partner.",
+            example = "BPNL2222222222RR", required = true) @RequestParam() String partnerBpnl,
```

Comment:
Duplicate of annotations "Parameter" and "RequestParam"

Reply:
Yes, there are some ugly redundancies...
I will do some cleaning up